src/agent/application.cpp

```diff
@@ -68,9 +68,6 @@ Application::Application(const std::string               &aInterfaceName,
 #if OTBR_ENABLE_MDNS
     , mPublisher(Mdns::Publisher::Create([this](Mdns::Publisher::State aState) { this->HandleMdnsState(aState); }))
 #endif
-#if OTBR_ENABLE_DBUS_SERVER && OTBR_ENABLE_BORDER_AGENT
```

Comment:
Let's still put the construction of `DBusAgent` in initializer list. And let's first add ePSK feature only for RCP. We can change in this way:
1. Do not change `mDBusAgent` here.
2. Add a parameter to `DBusAgent::Init`, which is a pointer to `BorderAgent`.
3. In Application::InitRcpMode, pass `mBorderAgent to` `mDBusAgent.Init()`.
4. In `DBusAgent::Init`, in the case `OT_COPROCESSOR_RCP`, de-reference the pointer and pass it to the constructor of `DBusThreadObjectRcp`.

Then `DBusThreadObjectRcp` can have a reference to `BorderAgent`.

Reply:
Thanks! Modified according to the comments, could check again.